### ISSUE TITLE
CDN Version of schedule-x

### ISSUE THREAD

**Issue by scottwhitsitt** [schedule-x/schedule-x#429]
**Is your feature request related to a problem? Please describe.**  
For various reasons, we do not compile our Vue app and need to use CDN versions of libraries. Is there support for a CDN version that we can access?

**Describe the solution you'd like**  
The functionality is great, but we can only find a CDN version for the older qalendar. It seems that qalendar is not being supported so we're looking for a CDN version with similar functionality. The schedule-x would be a good fit for us if we can find a CDN version. 

**Additional context**  
Nothing to add

**Things that could speed up the process**
- [ ] I can picture implementing at least parts of this feature
- [ ] I believe I know how to do this, and am willing to implement the whole thing.

**Comment by tomosterlund** [comment]
Hi, have you tried unpkg?

https://www.unpkg.com/

I never uploaded Qalendar to any CDN, but there should be plenty of CDNs out there that upload everything from npm I believe.

Let me know if this helps :)

**Comment by scottwhitsitt** [comment]
@tomosterlund Apparently, we downloaded multiple files separately and had to piece them together to get things properly referenced. Do you have any intention of providing a CDN version that would be easier to install and maintain? I purchased a premium license and we'll give it a try.  

The below is from my CTO. 

In order to get the js file for schedule-x, we used jsdelivr which creates the umd version of the library:
https://cdn.jsdelivr.net/npm/@schedule-x/calendar@1.58.1/dist/

Then by looking at that umd file we can know what dependencies to also download (preact, jsxRuntime, hooks, compat, signals). We then fetch the umd version of each of those libraries and their dependencies and make sure the references between the modules were named correctly.

So if the extra paid features are packaged as additional modules or libraries we would just need the umd versions of each so we can add them to our loaded JS. I don't know exactly what jsdelivr does to generate the umd file from the npm package.

**Comment by tomosterlund** [comment]
Hi.

That is pretty impressive that you managed to make that happen! The premium plugins are structured in the same way: there are UMD modules, but just like I've discovered recently that the UMD modules of the open source repo need a bit of puzzling, so do the premium ones.

I'll have a look asap.

Also feel free to write me @tomosterlund at Discord if you need any help. I gladly do chat support for any premium users.

**Comment by tomosterlund** [comment]
A fix for this will be included in the next release (I'll try to make it happen tonight or over the weekend).

As of that release, you should then be able to interact with the calendar and the plugins like so:

```html
<!doctype html>
<html class="no-js" lang="">

<head>
  <script src="https://cdn.jsdelivr.net/npm/preact@10.23.2/dist/preact.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/preact@10.23.2/hooks/dist/hooks.umd.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/@preact/signals-core@1.8.0/dist/signals-core.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/@preact/signals@1.3.0/dist/signals.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/preact@10.23.2/jsx-runtime/dist/jsxRuntime.umd.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/preact@10.23.2/compat/dist/compat.umd.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/@schedule-x/calendar@1.59.0/dist/core.umd.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/@schedule-x/drag-and-drop@1.59.0/dist/core.umd.js"></script>

  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@schedule-x/theme-default@1.59.0/dist/index.css">
</head>

<body>
  <div class="calendar"></div>

  <script type="module">
    const { createCalendar, viewWeek } = window.SXCalendar;
    const { createDragAndDropPlugin } = window.SXDragAndDrop;

    const calendar = createCalendar({
      views: [viewWeek],
      events: [
        {
          id: '1',
          title: 'Event 1',
          start: '2024-08-23 09:00',
          end: '2024-08-23 10:00'
        },
      ],
      plugins: [createDragAndDropPlugin()]
    })

    calendar.render(document.querySelector('.calendar'))
  </script>
</body>
</html>
```

**Comment by tomosterlund** [comment]
Fixed in `@schedule-x/calendar@1.59.0`

The umd files of the premium packages now also work as of version `0.18.2`. Probably the fastest way to integrate them is via copying the umd-files into your application.

If you want them hosted via a domain-restricted CDN though, which of course would be better for getting updates, I could also set up some workflow to publish to a CDN parallel to the npm repo. I'd need to charge a bit for compute costs & setup if I do so though.
